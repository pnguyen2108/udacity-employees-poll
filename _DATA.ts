import type {IQuestionAnswer, IQuestionFormat, IQuestions} from "./src/models/questions.model";
import {formatQuestion} from "./src/helpers/control.helpers";
import type {IUser, IUsers} from "./src/models/users.model";
import {isNil as _isNil} from "lodash"

export let questions: IQuestions = {
    "8xf0y6ziyjabvozdd253nd": {
        id: "8xf0y6ziyjabvozdd253nd",
        author: "sarahedo",
        timestamp: 1467166872634,
        optionOne: {
            votes: ["sarahedo"],
            text: "have horrible short term memory"
        },
        optionTwo: {
            votes: [],
            text: "have horrible long term memory"
        }
    },
    "6ni6ok3ym7mf1p33lnez": {
        id: "6ni6ok3ym7mf1p33lnez",
        author: "johndoe",
        timestamp: 1468479767190,
        optionOne: {
            votes: [],
            text: "become a superhero"
        },
        optionTwo: {
            votes: ["johndoe", "sarahedo"],
            text: "become a supervillian"
        }
    },
    "am8ehyc8byjqgar0jgpub9": {
        id: "am8ehyc8byjqgar0jgpub9",
        author: "sarahedo",
        timestamp: 1488579767190,
        optionOne: {
            votes: [],
            text: "be telekinetic"
        },
        optionTwo: {
            votes: ["sarahedo"],
            text: "be telepathic"
        }
    },
    "loxhs1bqm25b708cmbf3g": {
        id: "loxhs1bqm25b708cmbf3g",
        author: "tylermcginnis",
        timestamp: 1482579767190,
        optionOne: {
            votes: [],
            text: "be a front-end developer"
        },
        optionTwo: {
            votes: ["sarahedo"],
            text: "be a back-end developer"
        }
    },
    "vthrdm985a262al8qx3do": {
        id: "vthrdm985a262al8qx3do",
        author: "tylermcginnis",
        timestamp: 1489579767190,
        optionOne: {
            votes: ["tylermcginnis"],
            text: "find $50 yourself"
        },
        optionTwo: {
            votes: ["johndoe"],
            text: "have your best friend find $500"
        }
    },
    "xj352vofupe1dqz9emx13r": {
        id: "xj352vofupe1dqz9emx13r",
        author: "johndoe",
        timestamp: 1493579767190,
        optionOne: {
            votes: ["johndoe"],
            text: "write JavaScript"
        },
        optionTwo: {
            votes: ["tylermcginnis"],
            text: "write Swift"
        }
    }
};

export let users: IUsers = {
    sarahedo: {
        id: "sarahedo",
        password: "Udacity123!",
        name: "Sarah Edo",
        avatarURL: "https://media.istockphoto.com/id/1331329483/vector/female-avatar-icon.jpg?s=612x612&w=0&k=20&c=VPS89ZjzR7Ft7YkJWCg2ItTQTeIhdXjEut-Hc1gcpPI=",
        answers: {
            "8xf0y6ziyjabvozdd253nd": "optionOne",
            "6ni6ok3ym7mf1p33lnez": "optionOne",
            "am8ehyc8byjqgar0jgpub9": "optionTwo",
            "loxhs1bqm25b708cmbf3g": "optionTwo"
        },
        questions: ["8xf0y6ziyjabvozdd253nd", "am8ehyc8byjqgar0jgpub9"]
    },
    tylermcginnis: {
        id: "tylermcginnis",
        password: "Udacity123!",
        name: "Tyler McGinnis",
        avatarURL: "https://cdn-icons-png.flaticon.com/512/147/147142.png",
        answers: {
            "vthrdm985a262al8qx3do": "optionOne",
            "xj352vofupe1dqz9emx13r": "optionTwo"
        },
        questions: ["loxhs1bqm25b708cmbf3g", "vthrdm985a262al8qx3do"]
    },
    johndoe: {
        id: "johndoe",
        password: "Udacity123!",
        name: "John Doe",
        avatarURL: "https://static.vecteezy.com/system/resources/thumbnails/006/487/917/small/man-avatar-icon-free-vector.jpg",
        answers: {
            "xj352vofupe1dqz9emx13r": "optionOne",
            "vthrdm985a262al8qx3do": "optionTwo",
            "6ni6ok3ym7mf1p33lnez": "optionOne"
        },
        questions: ["6ni6ok3ym7mf1p33lnez", "xj352vofupe1dqz9emx13r"]
    }
};

export const setLogin = (username: string, password: string) => {
    return new Promise<IUser>((resolve, reject) => {

        if (Object.keys(users).includes(username)) {
            const userData: IUser = users[username];

            if (userData.password !== password) {
                reject("Username or Password is incorrect");

                return;
            }

            return resolve(userData);

        } else {
            reject("Username or Password is incorrect");
        }
    });
};

export const _getAllUsers = () => {
    return new Promise((res) => {
        setTimeout(() => res({...users}), 1000);
    });
};

export const _getUserDetail = (userId: string) => {
    return new Promise((resolve, reject) => {
        setTimeout(() => {
            if (Object.keys(users).includes(userId)) {
                return resolve(users[userId]);
            } else {
                reject("User not found");
            }
        }, 1000);
    });
};

export const _getQuestions = () => {
    return new Promise((res) => {
        setTimeout(() => res({...questions}), 1000);
    });
};

export const _getQuestionDetail = (questionId: string) => {
    return new Promise((resolve, reject) => {
        setTimeout(() => {
            if (Object.keys(questions).includes(questionId)) {
                return resolve(questions[questionId]);
            } else {
                reject("Question not found");
            }
        }, 1000);
    });
};


export const _saveQuestion = (question: IQuestionFormat) => {
    return new Promise((res) => {
        const authedUser: string = question.author;

        if (!_isNil(users[authedUser])) {
            const formattedQuestion = formatQuestion(question);

            setTimeout(() => {
                questions = {
                    ...questions,
                    [formattedQuestion.id]: formattedQuestion
                };

                users = {
                    ...users,
                    [authedUser]: {
                        ...users[authedUser],
                        questions: users[authedUser]['questions'].concat([formattedQuestion.id])
                    }
                };

                res(questions);
            }, 1000);
        } else{
            res(questions);
        }
    });
};

export const _saveQuestionAnswer = ({authedUser, qid, answer}: IQuestionAnswer) => {
    return new Promise((resolve, reject) => {
        setTimeout(() => {
            if (!_isNil(users[authedUser])) {
                users = {
                    ...users,
                    [authedUser]: {
                        ...users[authedUser],
                        answers: {
                            ...users[authedUser].answers,
                            [qid]: answer
                        }
                    }
                };
            }

            if (!_isNil(questions[qid]) && !_isNil(users[authedUser])) {
                questions = {
                    ...questions,
                    [qid]: {
                        ...questions[qid],
                        [answer]: {
                            ...questions[qid][answer],
                            votes: questions[qid][answer].votes.concat([authedUser])
                        }
                    }
                };
            }

            resolve(questions)
        }, 500)
    })
};

