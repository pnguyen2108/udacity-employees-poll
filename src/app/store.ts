import { usersSlice } from "../store/usersSlice.store";
import { authSlice } from "../store/authSlice.store";
import type { Action, ThunkAction } from "@reduxjs/toolkit";
import { combineSlices, configureStore } from "@reduxjs/toolkit";
import { setupListeners } from "@reduxjs/toolkit/query";
import { questionsSlice } from "../store/questionsSlice.store";

const rootReducer = combineSlices(usersSlice, authSlice, questionsSlice);

export type RootState = ReturnType<typeof rootReducer>

export const makeStore = (preloadedState?: Partial<RootState>) => {
  const store = configureStore({
    reducer: rootReducer,
    preloadedState
  });

  setupListeners(store.dispatch);

  return store;
};

export const store = makeStore();

// Infer the type of `store`
export type AppStore = typeof store

// Infer the `AppDispatch` type from the store itself
export type AppDispatch = AppStore["dispatch"]

export type AppThunk<ThunkReturnType = void> = ThunkAction<
  ThunkReturnType,
  RootState,
  unknown,
  Action
>
