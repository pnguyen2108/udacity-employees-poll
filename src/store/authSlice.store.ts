import type { PayloadAction } from "@reduxjs/toolkit";
import { createAsyncThunk, createSlice } from "@reduxjs/toolkit";
import type { ILogin } from "../models/auth.model";
import type { IUser } from "../models/users.model";
import { setLogin } from "../../_DATA";

export type AuthState = {
  isAuthenticated: boolean,
  userData?: any,
  isLoading: boolean,
};

export const authInitialState: AuthState = {
  isAuthenticated: false,
  userData: null,
  isLoading: false
};

export const authLogin = createAsyncThunk(
  "auth/Login",
  async (loginData: ILogin) => {
    return await setLogin(loginData.username, loginData.password);
  }
);

export const authSlice = createSlice({
  name: "auth",
  initialState: authInitialState,
  reducers: {
    logout: (state: AuthState) => {

      return {
        ...state,
        isAuthenticated: false,
        userData: null
      };
    },
    setIsLoading: (state: AuthState, action: PayloadAction<boolean>) => ({
      ...state,
      isLoading: action.payload
    })
  },
  selectors: {
    getUserInformation: (authState) => authState.userData,
    getAuthenticated: (authState) => authState.isAuthenticated,
    getIsLoading: (authState) => authState.isLoading
  },
  extraReducers: (builder) => {
    builder
      .addCase(authLogin.pending, (state) => {
        state.isLoading = true;
      })
      .addCase(authLogin.fulfilled, (state, action: PayloadAction<IUser>) => {
        state.isAuthenticated = true;

        state.userData = action.payload;

        state.isLoading = false;

      })
      .addCase(authLogin.rejected, (state) => {

        state.isLoading = false;
      });
  }
});

export const { logout } = authSlice.actions;