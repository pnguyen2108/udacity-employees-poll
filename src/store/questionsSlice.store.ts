import { createAsyncThunk, createSlice } from "@reduxjs/toolkit";
import type { IQuestion, IQuestionAnswer, IQuestions } from "../modes/questions.model";
import { _getQuestionDetail, _getQuestions, _saveQuestionAnswer } from "../../_DATA";

export type QuestionState = {
  questions: IQuestions,
  questionDetail: IQuestion | null,
  status: string,
  loadingDetail: boolean
  loadingSaveAnswer: boolean
  loadingQuestions: boolean
};

const initialState: QuestionState = {
  questions: {},
  questionDetail: null,
  status: "empty",
  loadingQuestions: false,
  loadingDetail: false,
  loadingSaveAnswer: false
};

export const fetchQuestions = createAsyncThunk(
  "auth/fetchQuestions",
  async () => {
    return await _getQuestions();
  }
);

export const fetchQuestionDetail = createAsyncThunk(
  "auth/fetchQuestionDetial",
  async (questionId: string) => {
    return await _getQuestionDetail(questionId);
  }
);

export const setQuestionAnswer = createAsyncThunk(
  "auth/setQuestionAnswer",
  async (data: IQuestionAnswer) => {
    return await _saveQuestionAnswer(data);
  }
);


export const questionsSlice = createSlice({
  name: "questions",
  initialState,
  reducers: {},
  selectors: {
    getQuestions: (state) => {
      return state.questions;
    },
    getQuestionDetail: (state) => {
      return state.questions;
    }
  },
  extraReducers: (builder) => {
    // fetch question list
    builder
      .addCase(fetchQuestions.pending, (state: QuestionState) => {
        state.loadingQuestions = true;

        state.status = "empty";
      })
      .addCase(fetchQuestions.fulfilled, (state: QuestionState, action: any) => {

        state.questions = action.payload;

        state.status = "loaded";

        state.loadingQuestions = false;
      })
      .addCase(fetchQuestions.rejected, (state: QuestionState) => {
        state.loadingQuestions = false;

        state.status = "empty";
      });

    // fetch question detail
    builder
      .addCase(fetchQuestionDetail.pending, (state: QuestionState) => {
        state.loadingDetail = true;
      })
      .addCase(fetchQuestionDetail.fulfilled, (state: QuestionState, action: any) => {
        state.questionDetail = action.payload;

        state.loadingDetail = false;
      })
      .addCase(fetchQuestionDetail.rejected, (state: QuestionState) => {
        state.loadingDetail = false;
      });

    // set question answer
    builder
      .addCase(setQuestionAnswer.pending, (state: QuestionState) => {
        state.loadingSaveAnswer = true;
      })
      .addCase(setQuestionAnswer.fulfilled, (state: QuestionState, action: any) => {
        state.loadingDetail = false;
      })
      .addCase(setQuestionAnswer.rejected, (state: QuestionState) => {
        state.loadingSaveAnswer = false;
      });
  }
});
