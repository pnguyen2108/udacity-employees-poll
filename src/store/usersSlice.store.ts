import type { IUser } from "../modes/users.model";
import { createAsyncThunk, createSlice } from "@reduxjs/toolkit";
import { _getAllUsers, _getUserDetail } from "../../_DATA";

export type UsersState = {
  loadingUserDetail: boolean;
  loadingUsers: boolean;
  users: IUser[]
  userDetail: IUser | null
};

const initialState: UsersState = {
  loadingUserDetail: false,
  loadingUsers: false,
  users: [],
  userDetail: null
};

export const fetchUsers = createAsyncThunk(
  "auth/fetchUsers",
  async () => {
    return await _getAllUsers();
  }
);

export const fetchUserDetail = createAsyncThunk(
  "auth/fetchUserDetail",
  async (userId: string) => {
    return await _getUserDetail(userId);
  }
);


export const usersSlice = createSlice({
  name: "users",
  initialState,
  reducers: {},
  selectors: {
    getUsers: (state) => {
      return state.users;
    }

  },
  extraReducers: (builder) => {
    // fetch all user
    builder
      .addCase(fetchUsers.pending, (state: UsersState) => {
        state.loadingUsers = true;
      })
      .addCase(fetchUsers.fulfilled, (state: UsersState, action: any) => {
        state.users = Object.values(action.payload);
        state.loadingUsers = false;
      })
      .addCase(fetchUsers.rejected, (state: UsersState) => {
        state.loadingUsers = false;
      });
    // fetch user detail
    builder
      .addCase(fetchUserDetail.pending, (state) => {
        state.loadingUserDetail = true;
      })
      .addCase(fetchUserDetail.fulfilled, (state, action: any) => {
        state.userDetail = action.payload;

        state.loadingUserDetail = false;
      })
      .addCase(fetchUserDetail.rejected, (state) => {
        state.loadingUserDetail = false;
      });
  }
});

export const { getUsers } = usersSlice.selectors;