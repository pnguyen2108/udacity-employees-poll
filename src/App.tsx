import "./App.scss";
import "primereact/resources/themes/tailwind-light/theme.css";
import { BrowserRouter, Route, Routes } from "react-router-dom";
import "primeicons/primeicons.css";
import { HomePage } from "./pages/Home/Home.page";
import { LeaderboardPage } from "./pages/Leaderboard/Leaderboard.page";
import { AppLayoutPage } from "./pages/AppLayout/AppLayout.page";
import { AuthGuardComponent } from "./components/AuthGuard/AuthGuard.component";
import { LoginPage } from "./pages/Login/Login.page";
import { NotFoundPage } from "./pages/NotFound/NotFound.page";
import { NewQuestionPage } from "./pages/NewQuestion/NewQuestion.page";
import { useEffect } from "react";
import { fetchQuestions } from "./store/questionsSlice.store";
import { useAppDispatch } from "./app/hooks";
import { QuestionDetailPage } from "./pages/QuestionDetail/QuestionDetail.page";

const App = () => {
  const  dispatch = useAppDispatch();

  useEffect(() => {
    dispatch(fetchQuestions());
  }, []);

  return <BrowserRouter>
    <AuthGuardComponent>
      <Routes>
        <Route path="/login" element={<LoginPage />} />
        <Route element={<AppLayoutPage />}>
          <Route path="/" element={<HomePage />} />
          <Route path="/home" element={<HomePage />} />
          <Route path="/leaderboard" element={<LeaderboardPage />} />
          <Route path="/add" element={<NewQuestionPage />} />
          <Route path="/questions/:question_id" element={<QuestionDetailPage />} />
          <Route path="*" element={<NotFoundPage />} />
        </Route>
      </Routes>
    </AuthGuardComponent>
  </BrowserRouter>;
};
export default App;
