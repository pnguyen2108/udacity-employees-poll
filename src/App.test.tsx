import {describe, expect} from "vitest";
import {store} from "./app/store";
import {fetchUsers, userInitialState} from "./store/usersSlice.store";
import {questions, users} from "../_DATA";
import {authInitialState, authLogin, logout} from "./store/authSlice.store";
import {addQuestion, fetchQuestions, questionsInitialState, setQuestionAnswer} from "./store/questionsSlice.store";
import {IQuestionAnswer, IQuestionFormat} from "./modes/questions.model";
import {fireEvent, getByText, render} from "@testing-library/react";
import React from "react";
import {Provider} from "react-redux";
import App from "./App";

describe('App', () => {
    describe('Redux auth state test', () => {
        it('auth init state test', async () => {
            expect(store.getState().auth).toEqual(authInitialState)
        });

        it('login test', async () => {
            await store.dispatch(authLogin({
                username: users.sarahedo.id,
                password: users.sarahedo.password
            }));

            expect(store.getState().auth?.userData).toEqual(users.sarahedo)
        })

        it('logout test', () => {
            store.dispatch(logout())

            expect(store.getState().auth).toEqual(authInitialState)
        })
    })

    describe('Redux users state test', () => {
        it('user init state test', async () => {
            expect(store.getState().users).toEqual(userInitialState)
        });

        it('fetchUsers test', async () => {
            await store.dispatch(fetchUsers());

            expect(store.getState().users?.users).toEqual(Object.values(users))
        })
    })

    describe('Redux questions state test', () => {
        it('questions init state test', async () => {
            expect(store.getState().questions).toEqual(questionsInitialState)
        });

        it('fetchQuestions test', async () => {
            await store.dispatch(fetchQuestions());

            expect(store.getState().questions?.questions).toEqual(questions)
        })

        it('saveQuestionAnswer success test', async () => {
            const questionId = "am8ehyc8byjqgar0jgpub9"

            const question = questions[questionId]

            const answer: IQuestionAnswer = {
                qid: questionId,
                authedUser: users.johndoe.id,
                answer: 'optionTwo'
            }

            const currOptionTwoVotes = question.optionTwo.votes.length;

            await store.dispatch(setQuestionAnswer(answer))

            const stateQuestion = store.getState().questions?.questions[questionId]

            expect(stateQuestion.optionTwo.votes.length).toBeGreaterThan(currOptionTwoVotes)

        })

        it('saveQuestionAnswer failed test', async () => {
            const questionId = "am8ehyc8byjqgar0jgpub9"

            const question = questions[questionId]

            const answer: IQuestionAnswer = {
                qid: questionId,
                authedUser: '111',
                answer: 'optionTwo'
            }

            const currOptionTwoVotes = question.optionTwo.votes.length;

            await store.dispatch(setQuestionAnswer(answer))

            const stateQuestion = store.getState().questions?.questions[questionId]

            expect(stateQuestion.optionTwo.votes.length).toEqual(currOptionTwoVotes)

        })

        it('saveQuestion success test', async () => {
            const question: IQuestionFormat = {
                author: users.johndoe.id,
                optionOneText: 'optionOne',
                optionTwoText: 'optionTwo'
            }

            const currQuestionsLength = Object.keys(questions).length

            await store.dispatch(addQuestion(question))

            const stateQuestionsLength = Object.keys(store.getState().questions?.questions).length;

            expect(stateQuestionsLength).toBeGreaterThan(currQuestionsLength)
        })

        it('saveQuestion failed test', async () => {
            const question: IQuestionFormat = {
                author: '222',
                optionOneText: 'optionOne',
                optionTwoText: 'optionTwo'
            }

            const currQuestionsLength = Object.keys(questions).length

            await store.dispatch(addQuestion(question))

            const stateQuestionsLength = Object.keys(store.getState().questions?.questions).length;

            expect(stateQuestionsLength).toEqual(currQuestionsLength)
        })
    })

    describe('Render test', () => {
        it('init page render test', async () => {
            const initAppContainer = render(
                <React.StrictMode>
                    <Provider store={store}>
                        <App/>
                    </Provider>
                </React.StrictMode>
            );

            expect(initAppContainer).toMatchSnapshot();
        });

        it('fireEvent test', () => {
            const initAppContainer = render(
                <React.StrictMode>
                    <Provider store={store}>
                        <App/>
                    </Provider>
                </React.StrictMode>
            );

            const button = initAppContainer.getByText('Login')

            fireEvent.click(button)

            expect(initAppContainer.getByText('Login')).toBeDefined()
        });
    })
});