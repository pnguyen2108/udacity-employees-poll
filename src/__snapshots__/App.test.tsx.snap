// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > Render test > init page 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="bg-gray-200 h-screen flex justify-center items-center"
      >
        <div
          class="p-toast p-component p-toast-top-right p-ripple-disabled"
          data-pc-name="toast"
          data-pc-section="root"
          style="position: fixed; top: 20px; right: 20px;"
        >
          <div />
        </div>
        <form
          class="bg-white shadow-xl rounded-xl login-form w-[500px] h-[350px]"
        >
          <div
            class=" my-4 mx-auto w-[400px]"
          >
            <h3
              class="text-3xl  text-violet-600 text-center"
            >
              Employee Polls
            </h3>
            <div
              class="flex flex-col mt-5"
            >
              <label
                class="font-semibold mb-1"
              >
                Username
              </label>
              <input
                class="p-inputtext p-component border-gray-300 border rounded-md h-10 w-full px-2"
                data-pc-name="inputtext"
                data-pc-section="root"
                value=""
              />
            </div>
            <div
              class="flex flex-col mt-8"
            >
              <label
                class="font-semibold mb-1"
              >
                Password
              </label>
              <div
                class="p-password p-component p-inputwrapper"
                data-pc-name="password"
                data-pc-section="root"
              >
                <input
                  autocomplete="off"
                  class="p-inputtext p-component p-password-input"
                  data-pc-name="inputtext"
                  data-pc-section="root"
                  tabindex="1"
                  type="password"
                  value=""
                />
              </div>
            </div>
            <button
              class="w-full btn text-center h-10 flex items-center justify-center cursor-pointer
                              bg-violet-500 hover:bg-violet-600 rounded-md text-white ra mt-5"
              type="submit"
            >
              Login
            </button>
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="bg-gray-200 h-screen flex justify-center items-center"
    >
      <div
        class="p-toast p-component p-toast-top-right p-ripple-disabled"
        data-pc-name="toast"
        data-pc-section="root"
        style="position: fixed; top: 20px; right: 20px;"
      >
        <div />
      </div>
      <form
        class="bg-white shadow-xl rounded-xl login-form w-[500px] h-[350px]"
      >
        <div
          class=" my-4 mx-auto w-[400px]"
        >
          <h3
            class="text-3xl  text-violet-600 text-center"
          >
            Employee Polls
          </h3>
          <div
            class="flex flex-col mt-5"
          >
            <label
              class="font-semibold mb-1"
            >
              Username
            </label>
            <input
              class="p-inputtext p-component border-gray-300 border rounded-md h-10 w-full px-2"
              data-pc-name="inputtext"
              data-pc-section="root"
              value=""
            />
          </div>
          <div
            class="flex flex-col mt-8"
          >
            <label
              class="font-semibold mb-1"
            >
              Password
            </label>
            <div
              class="p-password p-component p-inputwrapper"
              data-pc-name="password"
              data-pc-section="root"
            >
              <input
                autocomplete="off"
                class="p-inputtext p-component p-password-input"
                data-pc-name="inputtext"
                data-pc-section="root"
                tabindex="1"
                type="password"
                value=""
              />
            </div>
          </div>
          <button
            class="w-full btn text-center h-10 flex items-center justify-center cursor-pointer
                              bg-violet-500 hover:bg-violet-600 rounded-md text-white ra mt-5"
            type="submit"
          >
            Login
          </button>
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`App > Render test > init page render test 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="bg-gray-200 h-screen flex justify-center items-center"
      >
        <div
          class="p-toast p-component p-toast-top-right p-ripple-disabled"
          data-pc-name="toast"
          data-pc-section="root"
          style="position: fixed; top: 20px; right: 20px;"
        >
          <div />
        </div>
        <form
          class="bg-white shadow-xl rounded-xl login-form w-[500px] h-[350px]"
        >
          <div
            class=" my-4 mx-auto w-[400px]"
          >
            <h3
              class="text-3xl  text-violet-600 text-center"
            >
              Employee Polls
            </h3>
            <div
              class="flex flex-col mt-5"
            >
              <label
                class="font-semibold mb-1"
              >
                Username
              </label>
              <input
                class="p-inputtext p-component border-gray-300 border rounded-md h-10 w-full px-2"
                data-pc-name="inputtext"
                data-pc-section="root"
                value=""
              />
            </div>
            <div
              class="flex flex-col mt-8"
            >
              <label
                class="font-semibold mb-1"
              >
                Password
              </label>
              <div
                class="p-password p-component p-inputwrapper"
                data-pc-name="password"
                data-pc-section="root"
              >
                <input
                  autocomplete="off"
                  class="p-inputtext p-component p-password-input"
                  data-pc-name="inputtext"
                  data-pc-section="root"
                  tabindex="1"
                  type="password"
                  value=""
                />
              </div>
            </div>
            <button
              class="w-full btn text-center h-10 flex items-center justify-center cursor-pointer
                              bg-violet-500 hover:bg-violet-600 rounded-md text-white ra mt-5"
              type="submit"
            >
              Login
            </button>
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="bg-gray-200 h-screen flex justify-center items-center"
    >
      <div
        class="p-toast p-component p-toast-top-right p-ripple-disabled"
        data-pc-name="toast"
        data-pc-section="root"
        style="position: fixed; top: 20px; right: 20px;"
      >
        <div />
      </div>
      <form
        class="bg-white shadow-xl rounded-xl login-form w-[500px] h-[350px]"
      >
        <div
          class=" my-4 mx-auto w-[400px]"
        >
          <h3
            class="text-3xl  text-violet-600 text-center"
          >
            Employee Polls
          </h3>
          <div
            class="flex flex-col mt-5"
          >
            <label
              class="font-semibold mb-1"
            >
              Username
            </label>
            <input
              class="p-inputtext p-component border-gray-300 border rounded-md h-10 w-full px-2"
              data-pc-name="inputtext"
              data-pc-section="root"
              value=""
            />
          </div>
          <div
            class="flex flex-col mt-8"
          >
            <label
              class="font-semibold mb-1"
            >
              Password
            </label>
            <div
              class="p-password p-component p-inputwrapper"
              data-pc-name="password"
              data-pc-section="root"
            >
              <input
                autocomplete="off"
                class="p-inputtext p-component p-password-input"
                data-pc-name="inputtext"
                data-pc-section="root"
                tabindex="1"
                type="password"
                value=""
              />
            </div>
          </div>
          <button
            class="w-full btn text-center h-10 flex items-center justify-center cursor-pointer
                              bg-violet-500 hover:bg-violet-600 rounded-md text-white ra mt-5"
            type="submit"
          >
            Login
          </button>
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
